src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
and it may looks unrequired DocumentImpl parameter, but it help a lot with debugging to make sure that document is right one (Personally, I do not like that index have to hold state: document and mode. That lead to bugs)